src/main/java/seedu/address/logic/command/copy/CopyCommand.java

```diff
@@ -57,6 +61,17 @@ public CommandResult execute(Model model) throws CommandException {
             Activity activityToCopy = filteredActivityList.get(activityIndex.getZeroBased());
             TravelPlan travelPlan = travelPlanList.get(travelPlanIndex.getZeroBased());
 
+            WanderlustDate travelPlanStartDate = travelPlan.getStartDate();
+            WanderlustDate travelPlanEndDate = travelPlan.getEndDate();
+            WanderlustDateTime activityDateTime = activityToCopy.getActivityDateTime();
+
+            boolean isValidActivityDateTime = model.isValidActivityDate(activityDateTime,
+                    travelPlanStartDate, travelPlanEndDate);
```

Comment:
I think the date can be abstracted out, just pass the TravelPlan to isValidActivityDate() then in Model, you get the start and end date there.

Reply:
Ok solved, thanks!